elide-integration-tests/test/java/com/yahoo/elide/endpoints/ResourceIT.java

```diff
@@ -1130,6 +1117,53 @@ public void patchExtBadDelete() {
             .body(equalTo(expected));
     }
 
+    @Test(priority = 30)
+    public void createParentWithoutId() {
+        String req = getJson("/ResourceIT/createParentWithoutId.req.json");
+        String res = getJson("/ResourceIT/createParentWithoutId.json");
+
+        String response = given()
+                .contentType("application/vnd.api+json")
+                .accept("application/vnd.api+json")
+                .body(req)
+                .post("/parent")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .extract().body().asString();
+
+        assertEqualDocuments(response, res);
+    }
+
+    @Test
+    public void assignedId() {
+        String req = getJson("/ResourceIT/assignedId.req.json");
+        String res = getJson("/ResourceIT/assignedId.json");
+
+        String response = given()
+                .contentType("application/vnd.api+json")
+                .accept("application/vnd.api+json")
+                .body(req)
+                .post("/assignedId")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .extract().body().asString();
+
+        assertEqualDocuments(response, res);
+    }
+
+    @Test
+    public void assignedIdWithoutProvidedId() {
+        String req = getJson("/ResourceIT/assignedIdWithoutId.req.json");
+
+        given()
+            .contentType("application/vnd.api+json")
+            .accept("application/vnd.api+json")
+            .body(req)
+            .post("/assignedId")
+            .then()
+            .statusCode(HttpStatus.SC_FORBIDDEN);
+    }
+
     @Test
```

Comment:
maybe have a test that tries to assign an id to something with the generated annotation?